replay-handler-cloudformation.yaml

```
@@ -178,58 +178,67 @@ Resources:
-      InstanceType: !Ref InstanceType
-      AssociatePublicIpAddress: "false"
-      UserData: !Base64 
-        'Fn::Join':
-          - ''
-          - - |
-              #!/bin/bash -xe
-              #
-              #
-              ### update and install ###
-              #
-              export HOME=~
-              yum update -y
-              yum install git -y
-              yum install go -y
-              # dependency of github.com/google/gopacket
-              yum install libpcap-devel -y
-              #
-              #
-              ### dependency of main.go ###
-              go env GOPATH
-              echo 'export GOPATH=$HOME/go' >>~/.bash_profile
-              source ~/.bash_profile
-              #
-              #
-              #
-              ### create a virtual network interface that gets decapsulated VXLAN packets
-              #### compile & run go script ###
-              go install github.com/aws-samples/http-requests-mirroring@latest
-            - 'Fn::Sub': |
-                sudo ip link add vxlan0 type vxlan id ${TrafficMirroringVNI} dev eth0 dstport 4789
-                sudo ip link set vxlan0 up
-                $GOPATH"/bin/http-requests-mirroring" -destination "${RequestsForwardDestination}" -percentage "${ForwardPercentage}" -percentage-by "${PercentageBy}" -percentage-by-header "${PercentageByHeader}" -filter-request-port "${FilterByDestinationPort}" ${KeepOriginalHostHeader}
+      LaunchTemplateName: LT-ReplayHandler
+      LaunchTemplateData:
+        KeyName: !If [KeyNameEmpty, !Ref 'AWS::NoValue', !Ref KeyName]
+        ImageId: !If [UseArm, !Ref LatestAmiIdArm, !Ref LatestAmiIdX86]
+        SecurityGroupIds:
+          - !Ref InstanceSecurityGroup
+        InstanceType: !Ref InstanceType
+        UserData: !Base64 
+          'Fn::Join':
+            - ''
+            - - |
+                #!/bin/bash -xe
+                #
+                #
+                ### update and install ###
+                #
+                export HOME=~
+                yum update -y
+                yum install git -y
+                yum install go -y
+                # dependency of github.com/google/gopacket
+                yum install libpcap-devel -y
+                #
+                #
+                ### dependency of main.go ###
+                go env GOPATH
+                echo 'export GOPATH=$HOME/go' >>~/.bash_profile
+                source ~/.bash_profile
+                #
+                #
+                #
+                ### create a virtual network interface that gets decapsulated VXLAN packets
+                #### compile & run go script ###
+                wget https://github.com/aws-samples/http-requests-mirroring/raw/main/main.go -P $GOPATH"/src/http-requests-mirroring"
+                pushd $GOPATH"/src/http-requests-mirroring"
+                go mod init http-requests-mirroring
+                go mod tidy
+                go install .
+                popd
```

do we need this block of code? if not, can we simplify to the suggested code?

```suggestion
                go install github.com/aws-samples/http-requests-mirroring@latest
```

---

First, go install github.com/aws-samples/http-requests-mirroring@latest and it will not give you a binary, so you'll need to take the suggested code and build it.